packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -775,6 +783,58 @@ - (void)invalidateLayer
     _filterLayer.zPosition = CGFLOAT_MAX;
     [self.layer addSublayer:_filterLayer];
   }
+  
+  [self clearExistingGradientLayers];
+  auto background = _props->background;
+  if (background.empty()) {
+    return;
+  }
+  for (const auto &gradientPrimitive : background) {
+    CAGradientLayer *gradientLayer = [CAGradientLayer layer];
+    NSMutableArray *colors = [NSMutableArray array];
+    NSMutableArray *locations = [NSMutableArray array];
+    for (const auto& colorStop : gradientPrimitive.colorStops) {
+      if (colorStop.position.has_value()) {
+        auto location = @(colorStop.position.value());
+        UIColor* color = RCTUIColorFromSharedColor(colorStop.color);
+        [colors addObject:(id) color.CGColor];
+        [locations addObject:location];
+      }
+    }
+    gradientLayer.startPoint = CGPointMake(gradientPrimitive.startX, gradientPrimitive.startY);
+    gradientLayer.endPoint = CGPointMake(gradientPrimitive.endX, gradientPrimitive.endY);
+    
+    if (locations.count > 0) {
+      gradientLayer.locations = locations;
+    }
+    gradientLayer.colors = colors;
+    gradientLayer.frame = layer.bounds;
+    
+    // So that border layer is always above gradient layer
+    gradientLayer.zPosition = _borderLayer.zPosition;
+    [self.layer insertSublayer:gradientLayer atIndex:0];
```

Comment:
is there a reason this is at index 0? The z position should take care of that dimension for the most part. If you want gradients defined first in the array to be layered on top of the latter ones, maybe consider iterating over the gradients in reverse order so that you can just append to the end here.

Reply:
Updated this to below. Tried to explain the approach here in a little bit [longer answer](https://github.com/facebook/react-native/pull/45434#issuecomment-2265119204)

```
// So that border layer is always above gradient layer
gradientLayer.zPosition = _borderLayer.zPosition - 1;
[self.layer addSublayer:gradientLayer];
```